lbaas-octavia/main.tf

```diff
@@ -0,0 +1,192 @@
+  name        = "sg_ssh"
+  description = "Allow inbound SSH"
+
+  rule {
+    from_port   = 22
+    to_port     = 22
+    ip_protocol = "tcp"
+    cidr        = "0.0.0.0/0"
+  }
+}
+
+resource "openstack_compute_secgroup_v2" "sg_web" {
+  name        = "sg_web"
+  description = "Allow inbound HTTP"
+
+  rule {
+    from_port   = 80
+    to_port     = 80
+    ip_protocol = "tcp"
+    cidr        = "0.0.0.0/0"
+  }
+}
+
+resource "openstack_networking_network_v2" "net_lbdemo" {
+  name           = "net_lbdemo"
+  admin_state_up = "true"
+}
+
+resource "openstack_networking_subnet_v2" "subnet_lbdemo" {
+  name            = "subnet_lbdemo"
+  network_id      = openstack_networking_network_v2.net_lbdemo.id
+  cidr            = "192.168.1.0/24"
+  dns_nameservers = ["37.123.105.116", "37.123.105.117"]
+  ip_version      = 4
+}
+
+resource "openstack_networking_router_v2" "router_lbdemo" {
+  name                = "router_lbdemo"
+  admin_state_up      = true
+  external_network_id = data.openstack_networking_network_v2.ext-net.id
+}
+
+resource "openstack_networking_router_interface_v2" "routerint_lbdemo" {
+  router_id = openstack_networking_router_v2.router_lbdemo.id
+  subnet_id = openstack_networking_subnet_v2.subnet_lbdemo.id
+}
+
+data "template_file" "cloud_config" {
+  template = file("${path.module}/assets/cloud.cfg")
+
+  vars = {
+    init_app_sh = base64encode(file("${path.module}/assets/init-app.sh"))
+  }
+}
+
+resource "openstack_compute_instance_v2" "instance_lbdemo" {
+  count       = 3
+  name        = "App Instance ${count.index + 1}"
+  image_id    = data.openstack_images_image_v2.image.id
+  flavor_name = "m1.tiny"
+  key_pair    = openstack_compute_keypair_v2.keypair.name
+  user_data   = data.template_file.cloud_config.rendered
+
+  security_groups = [
+    "default",
+    openstack_compute_secgroup_v2.sg_web.name,
+  ]
+
+  network {
+    uuid = openstack_networking_network_v2.net_lbdemo.id
+  }
+
+  lifecycle {
+    ignore_changes = [image_id]
+  }
+}
+
+resource "openstack_compute_instance_v2" "instance_jumphost" {
+  name        = "Jumphost"
+  image_id    = data.openstack_images_image_v2.image.id
+  flavor_name = "m1.tiny"
+  key_pair    = openstack_compute_keypair_v2.keypair.name
+
+  security_groups = [
+    "default",
+    openstack_compute_secgroup_v2.sg_ssh.name,
+  ]
+
+  network {
+    uuid = openstack_networking_network_v2.net_lbdemo.id
+  }
+
+  lifecycle {
+    ignore_changes = [image_id]
+  }
+}
+
+resource "openstack_compute_floatingip_v2" "fip_lbdemo_jumphost" {
+  pool = "ext-net"
+}
+
+resource "openstack_compute_floatingip_associate_v2" "fipas_lbdemo" {
+  floating_ip = openstack_compute_floatingip_v2.fip_lbdemo_jumphost.address
+  instance_id = openstack_compute_instance_v2.instance_jumphost.id
+}
+
+resource "openstack_lb_loadbalancer_v2" "lb_app" {
+  vip_subnet_id = openstack_networking_subnet_v2.subnet_lbdemo.id
+  name          = "application loadbalancer"
+
+  security_group_ids = [
```

Comment:
I think the security group on the loadbalancer is not necessary. It is enough if there is a security group for the backends.

Reply:
removed the security group here